ext/json/json.c

```diff
@@ -141,24 +141,35 @@ static PHP_MINFO_FUNCTION(json)
 }
 /* }}} */
 
-PHP_JSON_API int php_json_encode_ex(smart_str *buf, zval *val, int options, zend_long depth) /* {{{ */
+PHP_JSON_API int php_json_encode_ex(smart_str *buf, zval *val, int options, zend_long depth, zend_string *indent_str) /* {{{ */
 {
 	php_json_encoder encoder;
 	int return_code;
+	bool default_indent_used = 0;
+
+	if (indent_str == 0) {
```

Comment:
Personally I'd not accept a NULL parameter here, and push up the creating of the default string into the `php_json_encode()` API.

Reply:
Good suggestion! I have moved the default indentation `zend_string *` to the `php_json_decode` function.